src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.Windows.cs

```diff
@@ -42,68 +41,77 @@ public static Architecture OSArchitecture
+                                case Interop.Kernel32.ProcessorArchitecture.Processor_Architecture_ARM64:
+                                    s_osArch = (int)Architecture.Arm64;
+                                    break;
+                                case Interop.Kernel32.ProcessorArchitecture.Processor_Architecture_ARM:
+                                    s_osArch = (int)Architecture.Arm;
+                                    break;
+                                case Interop.Kernel32.ProcessorArchitecture.Processor_Architecture_AMD64:
+                                    s_osArch = (int)Architecture.X64;
+                                    break;
+                                case Interop.Kernel32.ProcessorArchitecture.Processor_Architecture_INTEL:
+                                    s_osArch = (int)Architecture.X86;
+                                    break;
+                            }
+                        }
                     }
                 }
 
-                Debug.Assert(s_osArch != null);
-                return s_osArch.Value;
+                Debug.Assert(s_osArch != -1);
+                return (Architecture)s_osArch;
             }
         }
 
         public static Architecture ProcessArchitecture
         {
             get
             {
-                lock (s_processLock)
+                Debug.Assert(sizeof(Architecture) == sizeof(int));
+
+                if (s_processArch == -1)
                 {
-                    if (null == s_processArch)
+                    lock (s_lock)
                     {
-                        Interop.Kernel32.SYSTEM_INFO sysInfo;
-                        Interop.Kernel32.GetSystemInfo(out sysInfo);
-
-                        switch ((Interop.Kernel32.ProcessorArchitecture)sysInfo.wProcessorArchitecture)
+                        if (s_processArch == -1)
                         {
-                            case Interop.Kernel32.ProcessorArchitecture.Processor_Architecture_ARM64:
-                                s_processArch = Architecture.Arm64;
-                                break;
-                            case Interop.Kernel32.ProcessorArchitecture.Processor_Architecture_ARM:
-                                s_processArch = Architecture.Arm;
-                                break;
-                            case Interop.Kernel32.ProcessorArchitecture.Processor_Architecture_AMD64:
-                                s_processArch = Architecture.X64;
-                                break;
-                            case Interop.Kernel32.ProcessorArchitecture.Processor_Architecture_INTEL:
-                                s_processArch = Architecture.X86;
-                                break;
+                            Interop.Kernel32.SYSTEM_INFO sysInfo;
+                            Interop.Kernel32.GetSystemInfo(out sysInfo);
+
+                            switch ((Interop.Kernel32.ProcessorArchitecture)sysInfo.wProcessorArchitecture)
```

Comment:
Not that it matters much, but in this case I think the syntax would be clearer:

```CSharp
s_processArch = (Interop.Kernel32.ProcessorArchitecture)sysInfo.wProcessorArchitecture switch
{
    Interop.Kernel32.ProcessorArchitecture.Processor_Architecture_ARM64 => (int)Architecture.Arm64,
    ...
};
```

Same for above as well.